### ISSUE TITLE
remote broker -check_version problem

### ISSUE THREAD

**Issue by deepsrm** [dpkp/kafka-python#538]
Hi, 

I'm getting this error when trying to use the KafkaProducer in my production env. The kafka broker is 0.8.2.0

```
DEBUG:kafka.client:Node 0 connection state is <connected>
Traceback (most recent call last):
  kafka_producer = KafkaProducer(bootstrap_servers=['my_server:9092'])
  File "build/bdist.linux-x86_64/egg/kafka/producer/kafka.py", line 245, in __init__
  File "build/bdist.linux-x86_64/egg/kafka/client_async.py", line 625, in check_version
AssertionError
Exception AttributeError: "'KafkaProducer' object has no attribute '_closed'" in <bound method KafkaProducer.__del__ of <kafka.producer.kafka.KafkaProducer object at 0x7f66dc145d10>> ignored
```

**Comment by deepsrm** [comment]
Seems only happening when connecting outside, dont know why cant get the vheck version property

**Comment by deepsrm** [comment]
if i force the use of the api_version = '0.8.2' in the kafka producer it works, the problem is when using the auto feature by default...

**Comment by dpkp** [comment]
That is correct.

**Comment by dpkp** [comment]
That's a python internal thing that you should be able to safely ignore. I
can do some cleanup to avoid the weird log entry. But are you seeing actual
performance bugs? Messages not delivered? exception raised?

**Comment by deepsrm** [comment]
Once this exception is raised the producer stop working, if i force the version then it works normally, as i told this is only happening when connection to the broker outside. Python version 2.7.6

**Comment by dpkp** [comment]
I've updated the check_version code to be a bit more forgiving. Fix is in master @ `4d17ec2`
